lib/dependency-manager-adapters/npm.js

```diff
@@ -36,22 +36,22 @@ module.exports = CoreObject.extend({
   changeToDependencySet(depSet) {
     let adapter = this;
 
-    adapter.applyDependencySet(depSet);
-
-    return adapter._install(depSet).then(() => {
-      let deps = extend({}, depSet.dependencies || {}, depSet.devDependencies || {});
-      let currentDeps = Object.keys(deps).map((dep) => {
-        return {
-          name: dep,
-          versionExpected: deps[dep],
-          versionSeen: adapter._findCurrentVersionOf(dep),
-          packageManager: adapter.useYarnCommand ? 'yarn' : 'npm',
-        };
+    return adapter.applyDependencySet(depSet).then(() => {
```

Comment:
We absolutely should have been entangling this all along, this fix seems good regardless of the changes within `applyDependencySet` itself.